agent/consul/rpc.go

```diff
@@ -429,36 +454,55 @@ func (s *Server) raftApply(t structs.MessageType, msg interface{}) (interface{},
 // a snapshot.
 type queryFn func(memdb.WatchSet, *state.Store) error
 
+// we specify this as an interface so that we can allow a our legacy structs.QueryOptions
+// and the protobuf defined QueryOptions struct to both be used here.
+type queryOptions interface {
+	GetMinQueryIndex() uint64
+	GetMaxQueryTime() time.Duration
+	GetRequireConsistent() bool
+}
+
+// we specify this as an interface so that we can allow a our legacy structs.QueryMeta
+// and the protobuf defined QueryMeta struct to both be used here.
+type queryMeta interface {
+	SetLastContact(time.Duration)
+	SetKnownLeader(bool)
+	GetIndex() uint64
+	SetIndex(uint64)
+}
+
 // blockingQuery is used to process a potentially blocking query operation.
-func (s *Server) blockingQuery(queryOpts *structs.QueryOptions, queryMeta *structs.QueryMeta,
-	fn queryFn) error {
+func (s *Server) blockingQuery(queryOpts queryOptions, queryMeta queryMeta, fn queryFn) error {
 	var timeout *time.Timer
+	var queryTimeout time.Duration
 
+	minQueryIndex := queryOpts.GetMinQueryIndex()
 	// Fast path right to the non-blocking query.
-	if queryOpts.MinQueryIndex == 0 {
+	if minQueryIndex == 0 {
 		goto RUN_QUERY
 	}
 
+	queryTimeout = queryOpts.GetMaxQueryTime()
 	// Restrict the max query time, and ensure there is always one.
-	if queryOpts.MaxQueryTime > maxQueryTime {
-		queryOpts.MaxQueryTime = maxQueryTime
-	} else if queryOpts.MaxQueryTime <= 0 {
-		queryOpts.MaxQueryTime = defaultQueryTime
+	if queryTimeout > maxQueryTime {
+		queryTimeout = maxQueryTime
+	} else if queryTimeout <= 0 {
+		queryTimeout = defaultQueryTime
 	}
 
 	// Apply a small amount of jitter to the request.
-	queryOpts.MaxQueryTime += lib.RandomStagger(queryOpts.MaxQueryTime / jitterFraction)
+	queryTimeout += lib.RandomStagger(queryTimeout / jitterFraction)
```

Comment:
Should this updated value be updated on the original struct?

Reply:
No RPC endpoint that calls blocking query ever looks at the value after issuing the blocking query. In fact none of the RPC endpoints in the agent/consul package even look at the value but rather its passed through from being deserialized.

Reply:
So, to the best of my knowledge the answer is that we should not update the original.